lib/my_app.rb

```diff
@@ -0,0 +1,151 @@
+ENV['RACK_ENV'] ||= "development"
+
+require_relative '../data_mapper_setup.rb'
+require 'sinatra/base'
+require 'sinatra/flash'
+
+class MakersBnb < Sinatra::Base
+  use Rack::MethodOverride
+  register Sinatra::Flash
+
+  enable :sessions
+  set :session_secret, 'super secret'
+
+  helpers do
+    def current_user
+      @current_user ||= User.get(session[:user_id]) 
+    end
+  end
+
+  get '/' do
+    @spaces = Space.all
+    erb :index
+  end
+
+  post '/' do
+    a_space = Space.create(params)  
+    current_user.space << a_space
+    current_user.save
+    redirect to('/new')
+  end
+
+  get '/new' do 
+    erb :newPost
+  end
+
+  get '/users/new' do
+    @user = User.new
+    erb :new_user
+  end
+
+  post '/users' do
+    @user = User.create(email: params[:email],
+                        password: params[:password],
+                        password_confirmation: params[:password_confirmation],
+                        user_name: params[:user_name])
+    if @user.save
+      session[:user_id] = @user.id
+      redirect to '/'
+    else
+      flash.now[:errors] = @user.errors.full_messages
+      erb :new_user
+    end
+  end
+
+  get '/sessions/new' do
+    erb :'sessions/new'
+  end
+
+  get '/booked/:id' do 
+    space = Space.last(id: params[:id])
+    if current_user != nil
+      owner = space.user.last.user_name
+      user = current_user.user_name
+      @booking_object = Booking.create(name: user, owner: owner, room: space.name, confirmation: params[:false])
+      session[:booking_ex] = @booking_object.id
+      @booking = Booking.last(id: session[:booking_ex])
+    end
+      erb :confirmation
+  end
+
+  get '/booking-requests' do 
+    if current_user
+      @bookings = Booking.all(owner: current_user.user_name)
+    end
+    erb :booking_requests
+  end
+
+  get '/book' do 
+    @bookings_placed = Booking.all
+    erb :'book'
+  end
+
+
+  post '/sessions' do
+    user = User.authenticate(params[:email], params[:password])
+    if user
+      session[:user_id] = user.id
+      redirect to '/'
+    else
+      flash.now[:errors] = ['The email or password is incorrect']
+      erb :'sessions/new' 
+    end
+  end
+
+  delete '/sessions' do
+    session[:user_id] = nil
+    session[:booking_ex] = nil
+    flash.keep[:notice] = 'goodbye!'
+    redirect to '/'
+  end
+
+  get '/users/listings' do
+    if current_user != nil
+      @place = current_user.space
+    end
+    erb :user_listings
+  end
+
+  get '/confirm/:id' do 
+    @booking = Booking.last(id: params[:id])
+    @booking.update(confirmation: 'Booking confirmed')
```

Comment:
You shouldn't handle this sort of message in the model. Instead, how about a boolean (`@booking.update(confirmed: true)`)? Then put the `"Booking confirmed"` string somewhere in the view layer.

Also, this sort of frequent update, if it's going to happen a lot, is ripe for placing in the model. Wouldn't it be great if this whole controller action looked like this:

```ruby
@booking = Booking.last(id: params[:id]).confirm
redirect to '/booking-requests'
```

It'd be so easy to do: in `Booking`, define a `confirm` method that updates the right column and saves the booking. Wow, nice!